apps/web/modules/ee/two-factor-auth/components/disable-two-factor-modal.tsx

```diff
@@ -3,65 +3,71 @@
 import { getFormattedErrorMessage } from "@/lib/utils/helper";
 import { Button } from "@/modules/ui/components/button";
+import { FormControl, FormError, FormField, FormItem } from "@/modules/ui/components/form";
 import { Input } from "@/modules/ui/components/input";
 import { Modal } from "@/modules/ui/components/modal";
 import { OTPInput } from "@/modules/ui/components/otp-input";
 import { PasswordInput } from "@/modules/ui/components/password-input";
+import { zodResolver } from "@hookform/resolvers/zod";
 import { useTranslations } from "next-intl";
-import { useRouter } from "next/navigation";
-import React, { useEffect, useState } from "react";
-import { Controller, SubmitHandler, useForm } from "react-hook-form";
+import React, { useState } from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
 import toast from "react-hot-toast";
-
-type TDisableTwoFactorFormState = {
-  password: string;
-  code: string;
-  backupCode?: string;
-};
-
-type TDisableTwoFactorModalProps = {
+import { z } from "zod";
+import { ZUserPassword } from "@formbricks/types/user";
+
+const ZDisableTwoFactorFormState = z
+  .object({
+    password: ZUserPassword,
+    code: z.string().optional(),
+    backupCode: z.string().optional(),
+  })
+  .refine((data) => data.code || data.backupCode, {
+    message: "Either code or backup code must be provided.",
+    path: ["code"],
+  });
+
+type TDisableTwoFactorFormState = z.infer<typeof ZDisableTwoFactorFormState>;
+
+interface DisableTwoFactorModalProps {
   open: boolean;
   setOpen: (open: boolean) => void;
-};
-
-export const DisableTwoFactorModal = ({ open, setOpen }: TDisableTwoFactorModalProps) => {
+}
+
+export const DisableTwoFactorModal = ({ open, setOpen }: DisableTwoFactorModalProps) => {
+  const form = useForm<TDisableTwoFactorFormState>({
+    defaultValues: {
+      password: "",
+      code: "",
+      backupCode: "",
+    },
+    resolver: zodResolver(ZDisableTwoFactorFormState),
+  });
   const t = useTranslations();
-  const router = useRouter();
-  const { handleSubmit, control, setValue } = useForm<TDisableTwoFactorFormState>();
   const [backupCodeInputVisible, setBackupCodeInputVisible] = useState(false);
 
-  useEffect(() => {
-    setValue("backupCode", "");
-    setValue("code", "");
-  }, [backupCodeInputVisible, setValue]);
-
-  const resetState = () => {
-    setBackupCodeInputVisible(false);
-    setValue("password", "");
-    setValue("backupCode", "");
-    setValue("code", "");
-    setOpen(false);
-  };
-
   const onSubmit: SubmitHandler<TDisableTwoFactorFormState> = async (data) => {
     const { code, password, backupCode } = data;
-
     const disableTwoFactorAuthResponse = await disableTwoFactorAuthAction({ code, password, backupCode });
-
     if (disableTwoFactorAuthResponse?.data) {
       toast.success(disableTwoFactorAuthResponse.data.message);
-
-      router.refresh();
-      resetState();
+      form.reset();
```

Comment:
I think we should leave the `router.refresh` function call here to sync the server state again